### PR TITLE
Save column in censor resources tab

### PR DESCRIPTION
**Pull request by jimmyh77**

### CODE REVIEW

**Comment by insraq** [comment]
I feel this involves quite a lot of changes for a relatively trivial feature. The API is perhaps more advanced than we need? If the goal is to persist the sorting state, then maybe `TableView` can accept an optional `sortingState` prop `{ column: number, asc: boolean }` and when it is mutated, the value is updated.

Also I don't think the `useEffect` is necessary? You can simply update the value in the `onClick` handler.

File: src/scripts/ui/StatisticsBuildingBody.tsx [hunk 1]
```diff
@@ -426,6 +426,9 @@ function BuildingTab({ gameState }: IBuildingComponentProps): React.ReactNode {
    );
 }
 
+// biome-ignore lint/style/useConst:
```
Comment (insraq) [inline src/scripts/ui/StatisticsBuildingBody.tsx:429]: This can be a const?

Comment (jimmyh77) [reply src/scripts/ui/StatisticsBuildingBody.tsx:429]: It can.  Updated.

**Comment by jimmyh77** [comment]
If you are ok with this.  I can add the saving of column states to other areas of the UI.

File: src/scripts/ui/StatisticsBuildingBody.tsx (hunk 1 above)
Comment (insraq) [reply src/scripts/ui/StatisticsBuildingBody.tsx:429]: Did you push?

**Comment by insraq** [comment]
> If you are ok with this. I can add the saving of column states to other areas of the UI.

Yes. Go ahead. Adding to "singleton" tables are trivial - you just need a variable to store it. However, for Market/Caranvasary, it's tricky as you need to store this per each tile? (like in a `Map`)

**Comment by jimmyh77** [comment]
> > If you are ok with this. I can add the saving of column states to other areas of the UI.
> 
> Yes. Go ahead. Adding to "singleton" tables are trivial - you just need a variable to store it. However, for Market/Caranvasary, it's tricky as you need to store this per each tile? (like in a `Map`)

I was just thinking the same sorting state could be used for all Markets for example.  Not sure each Market/Caravan/Warehouse needs to be saved individually.

File: src/scripts/ui/StatisticsBuildingBody.tsx (hunk 1 above)
Comment (jimmyh77) [reply src/scripts/ui/StatisticsBuildingBody.tsx]: Pushed now.

**Comment by insraq** [comment]
> > > If you are ok with this. I can add the saving of column states to other areas of the UI.
> > 
> > 
> > Yes. Go ahead. Adding to "singleton" tables are trivial - you just need a variable to store it. However, for Market/Caranvasary, it's tricky as you need to store this per each tile? (like in a `Map`)
> 
> I was just thinking the same sorting state could be used for all Markets for example. Not sure each Market/Caravan/Warehouse needs to be saved individually.

Yeah, we can start with that

**Pull request merged**